gremlin-groovy/src/main/java/org/apache/tinkerpop/gremlin/groovy/jsr223/GroovyTranslator.java

```diff
@@ -115,7 +116,8 @@ else if (object instanceof Bytecode)
         else if (object instanceof Traversal)
             return convertToString(((Traversal) object).asAdmin().getBytecode());
         else if (object instanceof String) {
-            return (((String) object).contains("\"") ? "\"\"\"" + object + "\"\"\"" : "\"" + object + "\"").replace("$", "\\$");
+            return (((String) object).contains("\"") ? "\"\"\"" + StringEscapeUtils.escapeJava((String) object) + "\"\"\"" : "\"" + StringEscapeUtils.escapeJava((String) object) + "\"")
+                    .replace("$", "\\$");
```

Comment:
since you used the "groovy" package for this, i was wondering if it still handles the "$" properly or if we still need the final `replace("$", "\\$")`? either way, could you please modify your tests to include a "$" character so that we can be sure that gets asserted?

Reply:
The final `replace("$", "\\$")` has to be preserved or else some other original tests will be failed.